prov/tcp/src/tcpx_progress.c

```diff
@@ -434,8 +434,31 @@ void tcpx_progress(struct util_ep *util_ep)
 
 static int tcpx_try_func(void *util_ep)
 {
-	/* nothing to do here. When endpoints
-	   have incoming data, cq drives progress*/
+	uint32_t events;
+	struct util_wait_fd *wait_fd;
+	struct tcpx_ep *ep;
+	int ret;
+
+	ep = container_of(util_ep, struct tcpx_ep, util_ep);
+	wait_fd = container_of(((struct util_ep *)util_ep)->rx_cq->wait,
+			       struct util_wait_fd,
+			       util_wait);
+
+	if (slist_empty(&ep->tx_queue) ^
+	    ep->send_ready_monitor) {
```

Comment:
bitwise xor?  That doesn't look right.

Reply:
Intent here is to modify events when they are not same. As long as send_ready_monitor is just 1 bit value, it would work.  Alternatively, I can just check for (in)equality here too like (slist_empty(&ep->tx_queue) ==  !ep->send_ready_monitor)

Reply:
I wouldn't assume that slist_empty and send_ready_monitor have the same range of values, unless you define both as bool.  I would just restructure the code to something like:
```
if (!slist_empty() && ep->send_ready_monitor) {
    blah;
} else if (slist_empty() && !ep->send_ready_monitor) {
    blahblah;
}
```
This may duplicate a couple lines of code, but it's clear